substrate/frame/support/test/tests/pallet_ui/call_feeless_invalid_closure_arg3.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0308]: mismatched types
+  --> tests/pallet_ui/call_feeless_invalid_closure_arg3.rs:32:31
+   |
+31 |         #[pallet::feeless_if(|_: &OriginFor<T>, _s: &u32| -> bool { true })]
+   |                              --------------------------------------------- arguments to this function are incorrect
+32 |         pub fn foo(_: OriginFor<T>, _something: u64) -> DispatchResult { Ok(()) }
+   |                                     ^^^^^^^^^^ expected `&u32`, found `&u64`
+   |
+   = note: expected reference `&u32`
+              found reference `&u64`
```

Comment:
This is confusing, not quite what I'd expect.

I'd expect something more like

```suggestion
32 |         pub fn foo(_: OriginFor<T>, _something: u64) -> DispatchResult { Ok(()) }
   |                                     ^^^^^^^^^^ expected `u32`, found `u64`
   |
   = note: expected reference `u32`
              found reference `u64`
```

I wonder if it's feasible? 

Even better if we could get the error on the `feeless_if` args, since the call args are more likely to be the source of truth.

Reply:
As per the suggested syntax from @gavofyork,`feeless_if` should accept references to the arguments. But I can try to make the error span a bit better.

Reply:
This compile error asking for refs is pointing to the call args, not `feeless_if` args

Reply:
Done